converters/vol2heightfield.cpp

```diff
@@ -0,0 +1,181 @@
+/**
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ **/
+/**
+ * @file vol2heightfield.cpp
+ * @ingroup converters
+ * @author Bertrand Kerautret (\c kerautre@loria.fr )
+ * LORIA (CNRS, UMR 7503), University of Nancy, France
+ *
+ * @date 2015/03/18
+ *
+ * 
+ *
+ * This file is part of the DGtalTools.
+ */
+
+///////////////////////////////////////////////////////////////////////////////
+#include <iostream>
+#include <fstream>
+#include "DGtal/base/Common.h"
+#include "DGtal/helpers/StdDefs.h"
+#include "DGtal/images/ImageContainerBySTLVector.h"
+#include "DGtal/io/writers/GenericWriter.h"
+#include "DGtal/io/readers/VolReader.h"
+#include "DGtal/images/ConstImageAdapter.h"
+#include "DGtal/kernel/BasicPointFunctors.h"
+
+
+#include <boost/program_options/options_description.hpp>
+#include <boost/program_options/parsers.hpp>
+#include <boost/program_options/variables_map.hpp>
+
+using namespace std;
+using namespace DGtal;
+
+
+///////////////////////////////////////////////////////////////////////////////
+namespace po = boost::program_options;
+
+int main( int argc, char** argv )
+{
+  typedef ImageContainerBySTLVector < Z3i::Domain, unsigned char > Image3D;
+  typedef ImageContainerBySTLVector < Z2i::Domain, unsigned char> Image2D;
+  typedef DGtal::ConstImageAdapter<Image3D, Z2i::Domain, DGtal::functors::Point2DEmbedderIn3D<DGtal::Z3i::Domain>,
+                                   Image3D::Value,  DGtal::functors::Identity >  ImageAdapterExtractor;
+
+  // parse command line ----------------------------------------------
+  po::options_description general_opt("Allowed options are: ");
+  general_opt.add_options()
+    ("help,h", "display this message")
+    ("input,i", po::value<std::string>(), "volumetric file (.vol) " )
+    ("output,o", po::value<std::string>(), "sequence of discrete point file (.sdp) ") 
+    ("thresholdMin,m", po::value<int>()->default_value(128), "min threshold (default 128)" )
+    ("thresholdMax,M", po::value<int>()->default_value(255), "max threshold (default 255)" )
+    ("nx", po::value<double>()->default_value(0), "choose x comp of the normal vector." )
```

Comment:
set the x component of the projection direction
(and similarly for the other options?)